source/locationHelper.py

```diff
@@ -0,0 +1,402 @@
+#locationHelper.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Babbage B.V.
+
+"""Classes and helper functions for working with rectangles and coordinates."""
+
+from collections import namedtuple
+import windowUtils
+import winUser
+from ctypes.wintypes import RECT, SMALL_RECT, POINT
+import textInfos
+import wx
+
+class Point(namedtuple("Point",("x","y"))):
+	"""Represents a point on the screen."""
+
+	def __add__(self,other):
+		if not isinstance(other,_POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x+other.x),(self.y+other.y))
+
+	def __radd__(self,other):
+		return self.__add__(other)
+
+	def __lt__(self, other):
+		"""
+		Returns whether self is less than other.
+		This first compares y, than x coordinates.
+		For example: (x=4,y=3) < (x=3,y=4) because self.y is less than other.y.
+		To compare in opposite order (i.e. compare x, than y), use tuple(self) < tuple(other)
+		"""
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return (self.y, self.x) < (other.y, other.x)
+
+	def __le__(self, other):
+		"""
+		Returns whether self is less than or equal to other.
+		This first compares y, than x coordinates.
+		For example: (x=4,y=3) < (x=3,y=4) because self.y is less than other.y.
+		To compare in opposite order (i.e. compare x, than y), use tuple(self) <= tuple(other)
+		"""
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return (self.y, self.x) <= (other.y, other.x)
+
+	def __gt__(self, other):
+		"""
+		Returns whether self is greater than other.
+		This first compares y, than x coordinates.
+		For example: (x=3,y=4) > (x=4,y=3) because self.y is greater than other.y.
+		To compare in opposite order (i.e. compare x, than y), use tuple(self) > tuple(other)
+		"""
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return (self.y, self.x) > (other.y, other.x)
+
+	def __ge__(self, other):
+		"""
+		Returns whether self is greater than or equal to other.
+		This first compares y, than x coordinates.
+		For example: (x=3,y=4) > (x=4,y=3) because self.y is greater than other.y.
+		To compare in opposite order (i.e. compare x, than y), use tuple(self) >= tuple(other)
+		"""
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return (self.y, self.x) >= (other.y, other.x)
+
+	def __eq__(self, other):
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return self.x==other.x and self.y==other.y
+
+	def __neq__(self, other):

```

Comment:
Instead of almost completely duplicating the code, you can also do `return not self.__eq__(other)`. IMO this is a nice, compact way to express the logic of this function.